coffee/shepherd.coffee

```diff
@@ -117,6 +117,9 @@ class Step extends Evented
     @tether = new Tether extend(tetherOpts, @options.tetherOptions)
 
   show: =>
+      @options.beforeShowPromise?()?.then(=> @_show()) ? @_show()
```

Comment:
Could we keep the fat arrow please? Doing so wraps it in an anonymous function, and defines ```_this``` which keeps the context correct. Without it, my use case does not work. Is it detrimental to the codebase in some way if we keep it?